src/stim/simulators/tableau_simulator.cc

```diff
@@ -715,11 +723,10 @@ void TableauSimulator::Z(const OperationData &target_data) {
     }
 }
 
-simd_bits<MAX_BITWORD_WIDTH> TableauSimulator::sample_circuit(const Circuit &circuit, std::mt19937_64 &rng, int8_t sign_bias) {
-    TableauSimulator sim(rng, circuit.count_qubits(), sign_bias);
-    sim.expand_do_circuit(circuit);
+simd_bits<MAX_BITWORD_WIDTH> TableauSimulator::sample_circuit(const Circuit &circuit) {
```

Comment:
This method is no longer doing what it used to do. It used to give a sample from the given circuit. Now it runs the given circuit and then produces a sample. What if the simulator has already applied operations? Then it's not producing samples from the circuit. It will even include measurements run before the circuit was given.

Change this back to its old signature. Make it behave the same as it used to by moving the rng over to the simulator, and then moving it back into the argument at the end.

Reply:
Yes, it is a significant change in behavior. It seemed like what we might want to do if we are to move away from the world of shared RNGs. The previous API makes the user manage their RNGs (and hides the sims). The new one makes the user manage their sims (and hides the RNGs in the sims).

I suppose we're not moving away from the world of shared RNGs into the world of sims owning their RNGs, but instead into the world of sims that borrow and reuse RNGs as needed. Sorry, I misunderstood your intention earlier.

(Will address this and other comments soon.)